test/sys/test_socket.rs

```diff
@@ -72,6 +72,132 @@ pub fn test_timestamping() {
     assert!(std::time::Duration::from(diff).as_secs() < 60);
 }
 
+#[cfg(target_os = "freebsd")]
+#[cfg_attr(qemu, ignore)]
+#[test]
+pub fn test_timestamping_realtime() {
+    use nix::sys::socket::{
+        recvmsg, sendmsg, setsockopt, socket, sockopt::ReceiveTimestamp,
+        sockopt::TsClock, ControlMessageOwned, MsgFlags, SockFlag, SockType,
+        SockaddrIn, SocketTimestamp,
+    };
+    use std::io::{IoSlice, IoSliceMut};
+
+    let sock_addr = SockaddrIn::from_str("127.0.0.1:6790").unwrap();
```

Comment:
This port is used by other tests.  I suggest
```suggestion
    let sock_addr = SockaddrIn::from_str("127.0.0.1:6792").unwrap();
```